### PR TITLE
proof sanitization

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
Improve RUs for `DecimalComparisonSafe` and `CallSafe`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shaobo-he-aws** [comment]
It seems it duplicates functionalities in #56

File: cedar-dafny/validation/thm/model.dfy [hunk 1]
```diff
@@ -758,7 +758,16 @@ module validation.thm.model {
     requires IsDecimalComparisonName(name)
     requires ExtensionFunSafeRequires(name, args)
     ensures ExtensionFunSafeEnsures(name, args)
-  {}
+  {
```
Comment (andrewmwells-amazon) [inline cedar-dafny/validation/thm/model.dfy:761]: This is greater than 10M according to my VSCode and the internal dashboard but doesn't show up in https://github.com/cedar-policy/cedar-spec/pull/56 for some reason?

**Comment by andrewmwells-amazon** [comment]
> It seems it duplicates functionalities in #56
Agree for `CallSafe`, but I don't understand why `DecimalComparisonSafe` isn't showing up in that PR. I can rebase once John's PR is merged.

**Comment by shaobo-he-aws** [comment]
> > It seems it duplicates functionalities in #56
> 
> Agree for `CallSafe`, but I don't understand why `DecimalComparisonSafe` isn't showing up in that PR. I can rebase once John's PR is merged.

Probably because of the discrepancy of RU costs with different Dafny flags? The set of flags in the Makefile is different from VSCode's.

File: cedar-dafny/validation/thm/model.dfy (hunk 1 above)
Comment (khieta) [reply cedar-dafny/validation/thm/model.dfy:761]: The internal dashboard is (currently) based on the internal code, so there may be differences there. As far as the difference between VSCode and the metric used for #56 (which is outlined under "Checking Proof Stability" in the README) -- likely just due to different default settings between the command line build and VSCode. 

@emina may know more.

**Review by khieta — APPROVED** [review]

File: cedar-dafny/validation/thm/model.dfy (hunk 1 above)
Comment (emina) [reply cedar-dafny/validation/thm/model.dfy:761]: That and there are some differences between RU counts between VSCode and the command line even when the same flags are used.  We should use the command line counts as the ground truth.

**Pull request closed without merging**